src/toil/test/utils/toilDebugTest.py

```diff
@@ -0,0 +1,112 @@
+        self.tempDir = self._createTempDir(purpose='tempDir')
+
+    def tearDown(self):
+        """Default tearDown for unittest."""
+
+        shutil.rmtree(self.jobStoreDir)
+        ABC = os.path.abspath('src/toil/test/utils/ABC/ABC.txt')
+        if os.path.exists(ABC):
+            os.remove(ABC)
+
+        unittest.TestCase.tearDown(self)
+
+    @slow
+    def testJobStoreContents(self):
+        """Test toilDebugFile.printContentsOfJobStore().
+
+        Runs a workflow that imports 'B.txt' and 'mkFile.py' into the
+        jobStore.  'A.txt', 'C.txt', 'ABC.txt' are then created.  This checks to
+        make sure these contents are found in the jobStore and printed."""
+
+        contents = ['A.txt', 'B.txt', 'C.txt', 'ABC.txt', 'mkFile.py']
+
+        subprocess.check_call(['python', os.path.abspath('src/toil/utils/toilDebugFile.py'), self.jobStoreDir, '--listFilesInJobStore=True'])
+        jobstoreFileContents = os.path.abspath('jobstore_files.txt')
+        files = []
+        match = 0
+        with open(jobstoreFileContents, 'r') as f:
+            for line in f:
+                files.append(line.strip())
+        for file in files:
+            for expected_file in contents:
+                if file.endswith(expected_file):
+                    match = match + 1
+        logger.info(files)
+        logger.info(contents)
+        logger.info(match)
+        # C.txt will match twice (once with 'C.txt', and once with 'ABC.txt')
+        assert match == 6
+        os.remove(jobstoreFileContents)
+
+    # expected run time = 4s
+    def testFetchJobStoreFiles(self):
+        """Test toilDebugFile.fetchJobStoreFiles() without using symlinks."""
+        self.fetchFiles(symLink=False)
+
+    # expected run time = 4s
+    def testFetchJobStoreFilesWSymlinks(self):
+        """Test toilDebugFile.fetchJobStoreFiles() using symlinks."""
+        self.fetchFiles(symLink=True)
+
+    def fetchFiles(self, symLink):
+        """
+        Fn for testFetchJobStoreFiles() and testFetchJobStoreFilesWSymlinks().
+
+        Runs a workflow that imports 'B.txt' and 'mkFile.py' into the
+        jobStore.  'A.txt', 'C.txt', 'ABC.txt' are then created.  This test then
+        attempts to get a list of these files and copy them over into ./src from
+        the jobStore, confirm that they are present, and then delete them.
+        """
+        contents = ['A.txt', 'B.txt', 'C.txt', 'ABC.txt', 'mkFile.py']
+        outputDir = os.path.abspath('src')
+        cmd = ['python', os.path.abspath('src/toil/utils/toilDebugFile.py'),
+               self.jobStoreDir,
+               '--fetch', '*A.txt', '*B.txt', '*C.txt', '*ABC.txt', '*mkFile.py',
+               '--localFilePath=' + os.path.abspath('src'),
+               '--useSymlinks=' + str(symLink)]
+        subprocess.check_call(cmd)
+        for file in contents:
+            matchingFilesFound = recursiveGlob(outputDir, '*' + file)
+            assert len(matchingFilesFound) >= 1, matchingFilesFound
```

Comment:
nitpick: using `self.assertGreaterEqual()` could give a more informative error message should this test fail.

Reply:
This is pretty cool.  Thanks for showing this to me.